### ISSUE TITLE
Lab2 review Luca Marcellino

### ISSUE THREAD

**Issue by LucaMarcellino** [LorenzoRadaele/Computational-Intelligence#2]
# Review Luca Marcellino

The code is well wrote and the division between functions and code is well done. Minor review: please put some more comment also in the code to run and not only to the functions. 

I want to give you only a couple of suggestions:

- Try to choose a method that can change the mutation rate during the code: for example if you are stack in a minimum.
- Try to remove duplicates. If I well understood your code, you took 10 individuals and between them you took two form a tournament. But you don't delete the duplicates, so after some time, maybe, you have alle individual equal and make crossover with them. Probably this is the reason because you need a high mutation rate, because instead you stay a lot of time stack.
- Try maybe to change also the crossover, for example I used the two point crossover, I'm not sure but maybe you can have some better solution

So to summarize my idea is try to make some evaluation of your parameters.

Overall the code is good and the idea of your fitness could be a big boost for yuor solutions.
